src/sentry/migrations/0165_create_notificationsetting_table.py

```diff
@@ -0,0 +1,147 @@
+                            ),
+                        ]
+                    ),
+                ),
+                (
+                    "value",
+                    sentry.db.models.fields.bounded.BoundedPositiveIntegerField(
+                        choices=[
+                            (
+                                sentry.models.notificationsetting.NotificationSettingOptionValues(
+                                    0
+                                ),
+                                "default",
+                            ),
+                            (
+                                sentry.models.notificationsetting.NotificationSettingOptionValues(
+                                    10
+                                ),
+                                "off",
+                            ),
+                            (
+                                sentry.models.notificationsetting.NotificationSettingOptionValues(
+                                    20
+                                ),
+                                "on",
+                            ),
+                            (
+                                sentry.models.notificationsetting.NotificationSettingOptionValues(
+                                    30
+                                ),
+                                "subscribe_only",
+                            ),
+                            (
+                                sentry.models.notificationsetting.NotificationSettingOptionValues(
+                                    40
+                                ),
+                                "commited_only",
+                            ),
+                        ]
+                    ),
+                ),
+                (
+                    "organization",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        null=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        to="sentry.Organization",
+                    ),
+                ),
+                (
+                    "project",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        null=True, on_delete=django.db.models.deletion.CASCADE, to="sentry.Project"
+                    ),
+                ),
+                (
+                    "team",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        null=True, on_delete=django.db.models.deletion.CASCADE, to="sentry.Team"
+                    ),
+                ),
+                (
+                    "user",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        null=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        to=settings.AUTH_USER_MODEL,
+                    ),
+                ),
```

Comment:
Is it valid to have all of organization, project, team and user null?, or to have multiple columns set? If not you could use a check constraint to ensure at least one of the column is populated and another to prevent multiple columns being set if necessary.